### PR TITLE
Update devtools

### PR DESCRIPTION
**Pull request by white238**
Updates devtools on toss3 and blueos to really recent spack develop + my fixed cppcheck package which is sitting in a PR.

I spent enough time on this before I realized that we don't have toss4 devtools and that is going to have to be a Future Chris problem.

This also includes a new uberenv that has a non-backwards log.  It's amazing how long we lived with that.  Thanks @adrienbernede !

### CODE REVIEW

**Review by kennyweiss — APPROVED** [review]
Thanks @white238 !

**Review by bmhan12 — APPROVED** [review]
LGTM!

**Review by agcapps — APPROVED** [review]
Thank you for the update, @white238 .

File: scripts/spack/devtools_configs/blueos_3_ppc64le_ib_p9/packages.yaml [hunk 1]
```diff
@@ -15,121 +15,121 @@
...
+  # Lock in versions of Devtools
   cmake:
-    version: [3.14.5]
-    buildable: false
     externals:
-    - spec: cmake
-      prefix: /usr/tce/packages/cmake/cmake-3.14.5
+    - spec: cmake@3.18.0
+      prefix: /usr/tce/packages/cmake/cmake-3.18.0
+    buildable: false
```
Comment (bmhan12) [inline scripts/spack/devtools_configs/blueos_3_ppc64le_ib_p9/packages.yaml:128]: I'm curious: are just the devtool packages being built with cmake 3.18?

I see in the blueos host-config the header is still
`# CMake executable path: /usr/tce/packages/cmake/cmake-3.14.5/bin/cmake`

Comment (white238) [reply scripts/spack/devtools_configs/blueos_3_ppc64le_ib_p9/packages.yaml:128]: Yes this is only for the devtools.  Due to not wanting to rebuild the devtools everytime (it more than doubles the builds), we have to split the configs like this.  I really wish there was a better way but I don't know of one.

**Pull request merged**